src/python/pants/rules/core/test.py

```diff
@@ -25,19 +27,19 @@ class Test(Goal):
 @console_rule(Test, [Console, BuildFileAddresses])
 def fast_test(console, addresses):
   test_results = yield [Get(TestResult, Address, address.to_address()) for address in addresses]
-  wrote_any_stdout = False
   did_any_fail = False
-  for test_result in test_results:
-    wrote_any_stdout |= bool(test_result.stdout)
-    # Assume \n-terminated
-    console.write_stdout(test_result.stdout)
-    if test_result.stdout and not test_result.stdout[-1] == '\n':
-      console.write_stdout('\n')
-    if test_result.status == Status.FAILURE:
+  for address, test_result in zip(addresses, test_results):
+    if test_result.status == Status.SUCCESS:
```

Comment:
Is there a way this could read the "capture" flag? Because as a user, I find myself often wanting output even in successful runs.

Reply:
(Summarizing conversation we had over Slack)

No, it is not possible to capture the Pytest passthru arg `-s` that you were referring to, because this is a generic test runner and must support test runners other than Pytest.

Instead, we have to make the global decision of if we want to output stdout/stderr for success cases or not.

Pros
- Less verbose output, especially when running `./pants test ::`

Cons
- Lose information that may be helpful? E.g. maybe we want to see how many tests were ran for that target, if any were skipped, etc.
- Borja mentioned for Pantsd tests, it's very helpful to get the pantsd and exception logs printed at the end.

--

Happy to revert this change if you all decide it's worth keeping. I know the API too little to have a strong opinion here.

Reply:
I quite like Bazel's approach here: https://docs.bazel.build/versions/master/command-line-reference.html#flag--test_output - what you've implemented here is the `errors` value of that flag; it would be easy for us to implement all of those except for `streamed` (which we'd like to do, but have separate tickets for... Not super explicitly, but related to https://github.com/pantsbuild/pants/issues/6002)